File: 05-multiplication-table/mult_table.pl
# Perl program to generate multiplication tables
# Created by M.V.Harish Kumar on 09-01-2025
use v5.40;

sub gen_tables($n, $t) {
    say "$n Tables";
    say "$n x $_ = @{[$n*$_]}" for 1..$t;
}

print "Enter the no. to generate tables: ";
chomp(my $n = <STDIN>);
print "Enter the no. till which to generate: ";
chomp(my $t = <STDIN>);
gen_tables $n, $t;

File: 05-multiplication-table/mult_table.hs
-- Haskell program to print multiplication tables
-- Written by M.V.Harish Kumar on 11-07-2025
import System.IO (hFlush, stdout)
import Text.Printf (printf)

genTables :: Int -> Int -> IO ()
genTables n t = do
    printf "%d Tables\n\n" n
    mapM_ printLine [1..t]
    where
        printLine i = printf "%d x %d = %d\n" n i (n * i)

main = do
    putStr "Enter the no. to generate tables: "
    hFlush stdout
    n <- getLine

    putStr "Enter the no. till which to generate: "
    hFlush stdout
    t <- getLine

    genTables (read n) (read t)

